hassio/src/system/hassio-system-metrics.ts

```diff
@@ -0,0 +1,189 @@
+
+  @internalProperty() private _coreMetrics?: any;
+
+  protected render(): TemplateResult | void {
+    const usedSpace = this._getUsedSpace(this.hostInfo);
+
+    return html`
+      <ha-card header="System metrics">
+        <div class="card-content">
+          <ha-settings-row>
+            <span slot="heading">
+              Core CPU usage
+            </span>
+            <div slot="description">
+              <span class="value">
+                ${this._roundOrNull(this._coreMetrics?.cpu_percent)}%
+              </span>
+              <ha-bar
+                class="${classMap({
+                  target: this._coreMetrics?.cpu_percent > TARGET,
+                })}"
+                .value=${this._coreMetrics?.cpu_percent}
+              ></ha-bar>
+            </div>
+          </ha-settings-row>
+          <ha-settings-row>
+            <span slot="heading">
+              Core RAM usage
+            </span>
+            <div slot="description">
+              <span class="value">
+                ${this._roundOrNull(this._coreMetrics?.memory_percent)}%
+              </span>
+              <ha-bar
+                class="${classMap({
+                  target: this._coreMetrics?.memory_percent > TARGET,
+                })}"
+                .value=${this._coreMetrics?.memory_percent}
+              ></ha-bar>
+            </div>
+          </ha-settings-row>
+          <ha-settings-row>
+            <span slot="heading">
+              Supervisor CPU usage
+            </span>
+            <div slot="description">
+              <span class="value">
+                ${this._roundOrNull(this._supervisorMetrics?.cpu_percent)}%
+              </span>
+              <ha-bar
+                class="${classMap({
+                  target: this._supervisorMetrics?.cpu_percent > TARGET,
+                })}"
+                .value=${this._supervisorMetrics?.cpu_percent}
+              ></ha-bar>
+            </div>
+          </ha-settings-row>
+          <ha-settings-row>
+            <span slot="heading">
+              Supervisor RAM usage
+            </span>
+            <div slot="description">
+              <span class="value">
+                ${this._roundOrNull(this._supervisorMetrics?.memory_percent)}%
+              </span>
+              <ha-bar
+                class="${classMap({
+                  target: this._supervisorMetrics?.memory_percent > TARGET,
+                })}"
+                .value=${this._supervisorMetrics?.memory_percent}
+              ></ha-bar>
+            </div>
+          </ha-settings-row>
+          <ha-settings-row>
+            <span slot="heading">
+              Used space
+            </span>
+            <div slot="description">
+              <span class="value">${usedSpace}%</span>
+              <ha-bar
+                class="${classMap({ target: usedSpace > TARGET })}"
+                .value=${usedSpace}
+              ></ha-bar>
+            </div>
+          </ha-settings-row>
```

Comment:
Not sure what is in `_coreMetrics`, but can we do something like this?
```suggestion
          Object.values(this._coreMetrics).map(([key, value]) => html`<ha-settings-row>
            <span slot="heading">
              ${headings[key]}
            </span>
            <div slot="description">
              <span class="value">
                ${this._roundOrNull(value)}%
              </span>
              <ha-bar
                class="${classMap({
                  warning: value > WARNING,
                  critical: value > CRTITICAL,
                })}"
                .value=${value}
              ></ha-bar>
            </div>
          </ha-settings-row>
```

Reply:
My bad, I'll define a type for it...
```
blk_read: 30553088
blk_write: 16461824
cpu_percent: 4.99
memory_limit: 3938705408
memory_percent: 10.59
memory_usage: 416968704
network_rx: 0
network_tx: 0
```

Reply:
OK, let's go for the render function then 😄 

```ts
private _renderStats(description, value) {
return html`<ha-settings-row>
            <span slot="heading">
              ${description}
            </span>
            <div slot="description">
              <span class="value">
                ${this._roundOrNull(value)}%
              </span>
              <ha-bar
                class="${classMap({
                  warning: value > WARNING,
                  critical: value > CRTITICAL,
                })}"
                .value=${value}
              ></ha-bar>
            </div>
          </ha-settings-row>`
```